### ISSUE TITLE
OpenAPI generator - array type for property in schema not correctly generated

### ISSUE THREAD

**Issue by sven-petersen** [SAP/cloud-sdk-js#4618]
**Describe the bug**

There is a small error in the generated types affecting "nested array objects". E.g. when generating the type for an object of this shape:

```ts
{
  content: [{ name: "foo" }]
}
```

**To Reproduce**

1. Use this OpenAPI schema and save it to `minimal-example-api.yaml`:

```yaml
openapi: "3.0.3"
info:
  description: minimal example
  version: "0.0.1"
  title: "sample api"
paths:
  /my-path:
    get:
      summary: do something crazy
      operationId: myOperation
      responses:
        "200":
          description: "yay it worked"
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/MySchema"
components:
  schemas:
    MySchema:
      type: object
      required:
        - content
      properties:
        content:
          type: array
          items:
            type: object
            required:
              - name
            properties:
              name:
                type: string
```

2. Generate the OpenAPI client with

```bash
npm init -y && npm i @sap-cloud-sdk/openapi-generator@3.13.0
npx openapi-generator --input minimal-example-api.yaml --outputDir gen --overwrite
```

3. check the generated schema for `#/components/schemas/MySchema` which looks like this:

```ts
// ...
export type MySchema = {
  content: {
    name: string;
  } & Record<string, any>[];
} & Record<string, any>;
```

5. The type for `.content` is incorrect as `[]`/array only applies to `Record<string, any>` but should actually wrap the whole object type for the `content` property aka `{ name: string } & Record<string, any>`. So the full type should look like this:

```ts
// ...
export type MySchema = {
  // note the surrounding "( ... )" and that the [] is after the closing ")"
  content:  ( {
    name: string;
  } & Record<string, any> ) [];
} & Record<string, any>;

```

**Expected behavior**

The `content` should be properly typed as an array.

**Used Versions:**

- node version 20.11.1
- npm version 10.2.4
- SAP Cloud SDK version 3.13.0

**Impact / Priority**

The generated types are incorrect making it necessary to use `as any` or `as unknown` type casts defeating the purpose of this feature.

Affected development phase: e.g. Development

Impact: Impaired

**Comment by deekshas8** [comment]
Hi @sven-petersen ,

Thanks for raising this. We are working on a fix and will update you once it's available.

**Comment by deekshas8** [comment]
Hi @sven-petersen ,
We released a new version `3.14.0` with the fix.
